test/unit/org/apache/cassandra/locator/Ec2ConnectorTest.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.locator;
+
+import java.time.Duration;
+import java.util.Properties;
+
+import org.junit.Test;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.locator.Ec2MetadataServiceConnector.EC2MetadataType;
+import org.apache.cassandra.locator.Ec2MetadataServiceConnector.V1Connector;
+import org.apache.cassandra.locator.Ec2MetadataServiceConnector.V2Connector;
+
+import static java.lang.String.format;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.joining;
+import static org.apache.cassandra.locator.Ec2MetadataServiceConnector.DEFAULT_EC2_METADATA_URL;
+import static org.apache.cassandra.locator.Ec2MetadataServiceConnector.EC2_METADATA_TYPE_PROPERTY;
+import static org.apache.cassandra.locator.Ec2MetadataServiceConnector.V2Connector.AWS_EC2_METADATA_TOKEN_TTL_SECONDS_HEADER_PROPERTY;
+import static org.apache.cassandra.locator.Ec2MetadataServiceConnector.V2Connector.MAX_TOKEN_TIME_IN_SECONDS;
+import static org.apache.cassandra.locator.Ec2MetadataServiceConnector.V2Connector.MIN_TOKEN_TIME_IN_SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class Ec2ConnectorTest
+{
+    @Test
+    public void testV1Configuration()
+    {
+        Properties p = new Properties();
+        p.setProperty(EC2_METADATA_TYPE_PROPERTY, EC2MetadataType.v1.name());
+        Ec2MetadataServiceConnector ec2Connector = Ec2MetadataServiceConnector.create(new SnitchProperties(p));
+
+        assertTrue(ec2Connector instanceof V1Connector);
+
+        assertEquals(DEFAULT_EC2_METADATA_URL, ec2Connector.metadataServiceUrl);
+    }
+
+    @Test
+    public void testV2Configuration()
+    {
+        Ec2MetadataServiceConnector ec2Connector = Ec2MetadataServiceConnector.create(new SnitchProperties(new Properties()));
+
+        // v2 connector by default
+        assertTrue(ec2Connector instanceof V2Connector);
+        assertEquals(DEFAULT_EC2_METADATA_URL, ec2Connector.metadataServiceUrl);
+        assertEquals(Duration.ofSeconds(MAX_TOKEN_TIME_IN_SECONDS), ((V2Connector) ec2Connector).tokenTTL);
+    }
+
+    @Test
+    public void testInvalidConfiguration()
+    {
+        try
+        {
+            Properties p = new Properties();
+            p.setProperty(EC2_METADATA_TYPE_PROPERTY, "non-existent");
+
+            Ec2MetadataServiceConnector.create(new SnitchProperties(p));
+            fail("it should not be possible create a connector of type non-existent");
+        }
+        catch (ConfigurationException ex)
```

Comment:
Could you use AssertJ in tests? (see https://www.tabnine.com/code/java/methods/org.assertj.core.api.Assertions/assertThatExceptionOfType)

Reply:
rewritten